pkg/dependencies/factory.go

```diff
@@ -1669,7 +1669,15 @@ func (f *Factory) WithNutanixClientCache() *Factory {
 func getProxyConfiguration(clusterSpec *cluster.Spec) (httpProxy, httpsProxy string, noProxy []string) {
 	proxyConfiguration := clusterSpec.Cluster.Spec.ProxyConfiguration
 	if proxyConfiguration != nil {
-		return proxyConfiguration.HttpProxy, proxyConfiguration.HttpsProxy, proxyConfiguration.NoProxy
+		noProxyList := append(proxyConfiguration.NoProxy, clusterSpec.Cluster.Spec.ClusterNetwork.Pods.CidrBlocks...)
```

Comment:
Can you allocate a list like we do here so we are consistent? https://github.com/aws/eks-anywhere/blob/468a19a9b29d0bc7208630d980b7a6c5b757df17/pkg/clusterapi/proxy.go#L89